### ISSUE TITLE
Configuration File command line argument

### ISSUE THREAD

**Issue by ecc256** [locustio/locust#1381]
Ubuntu 18.04 + Locust 1.0.1
[Configuration File](https://docs.locust.io/en/latest/configuration.html#configuration-file)

with 2 config files in current folder:
`locust.conf`
`master.conf`

`locust --config=master.conf`
always use `locust.conf`, not `master.conf`

Not sure if it’s by design...?

**Comment by heyman** [comment]
That sounds like a bug, but I'm not able to reproduce it.

**Comment by heyman** [comment]
Tried with the following two configuration files in the current working dir:

### locust.conf
```
locustfile = ../basic.py
users = 1
hatch-rate = 1
run-time = 10s
headless = true
```

### master.conf
```
locustfile = ../basic.py
users = 4
hatch-rate = 1
run-time = 3s
headless = true
```

Running just `locust` sets the run time to 10 seconds, and running `locust --config=master` sets it to 3 seconds.

**Comment by ecc256** [comment]
Mine are:

**locust_gui.conf**
```
step-load
web-port=8080
only-summary
reset-stats
host=http://localhost
locustfile=test.py
```

**locust.conf**
```
headless
only-summary
reset-stats
host=http://localhost
users=16
hatch-rate=1
run-time=1m
locustfile=test.py
```


_locust --config locust_gui.conf_
starts **headless** test.

Anyway I can capture debug or something to prove?

**Comment by ecc256** [comment]
It looks like the sequence is:

1. default settings are set.
2. current folder **locust.conf** is read **if it exists** and overwrites default settings.
3. command line argument config is read **if was provided** (_--config locust_gui.conf_) and overwrites setting again.

Side questions:
If both files exist: **locust.conf** and **~/.locust.conf**, they will be used both, right?

**Comment by cyberw** [comment]
I think you may have encountered https://github.com/bw2/ConfigArgParse/issues/105

Unfortunately it is hard for us to fix, since it is an underlying issue in ConfigArgParse.

**Comment by heyman** [comment]
> Unfortunately it is not possible to clear the headless flag once it has been set because of an issue in configargparse bw2/ConfigArgParse#105

Oh, good catch. I actually didn't realize that the config files "stacked".

I guess we could consider removing the default locust configs (`./locust.conf`, `~/locust.conf`)? I don't see the big usecases for it, but I can definitely see people running in to issues (like this one) due to it.

**Comment by cyberw** [comment]
I use default configs every day (sometimes layered), it is super useful, so I dont want to remove it :)

But I guess an explicitly specified config file on the command line should not stack with the default ones.

**Comment by heyman** [comment]
> But I guess an explicitly specified config file on the command line should not stack with the default ones.

Yeah, maybe that's better. But how you you use layered configs then? Just the homedir and CWD?

**Comment by cyberw** [comment]
> > But I guess an explicitly specified config file on the command line should not stack with the default ones.
> 
> Yeah, maybe that's better. But how you you use layered configs then? Just the homedir and CWD?

Yup.

**Comment by ecc256** [comment]
Yep, layered configs are super useful!
Just want to add to Lars answer.
I use following in single config:
```
run-time=2s  # default
#run-time=20m
#run-time=10m
#run-time=5m
#run-time=2m
run-time=1m
#run-time=15s
#run-time=10s
#run-time=5s
```
I.e. if an option defined more than once, the last value is be used.
It can come from inherited or same config file.

Again, no need to remove anything.
But definitely need to add clarification to docs.
And about special headless flag too.

One more questions still:
I assume, the config inheritance is:
**defaults -> ~/locust.conf -> ./locust.conf -> cmd args**
Is it correct?

**Comment by cyberw** [comment]
Yes. And to be super specific, 

defaults -> ~/locust.conf -> ./locust.conf -> **env vars** -> cmd args

**Comment by ecc256** [comment]
Lars,
Super!
Could somebody add + clarify it [here](https://docs.locust.io/en/latest/configuration.html#configuration-file), please?
Specifically, 2nd paragraph.
